extensions/http-curl/client/HTTPStream.h

```diff
@@ -155,13 +155,13 @@ class HttpStream : public io::BaseStream {
  protected:
 
   /**
-   * Creates a vector and returns the vector using the provided
-   * type name.
+   * Populates the vector using the provided type name.
+   * @param buf output buffer
    * @param t incoming object
-   * @returns vector.
+   * @returns number of bytes read.
    */
   template<typename T>
-  std::vector<uint8_t> readBuffer(const T&);
+  int readBuffer(std::vector<uint8_t>&, const T&);
```

Comment:
I didn't check the code of BaseStream, just to confirm: the original function was declared at this level and it's not an override of a base func?

Reply:
Correct, it's an inline non-virtual member function that happens to be defined in many stream classes the same way. A lot of code duplication, but works.